examples/language-modeling/run_clm.py

```diff
@@ -340,25 +347,50 @@ def group_texts(examples):
         result["labels"] = result["input_ids"].copy()
         return result
 
+    def preprocess_function(examples):
+        examples = tokenizer(examples[text_column_name])
+        return group_texts(examples)
+
```

Comment:
This will not work grouped like this as the `group_texts` function relies on the length of tokenized samples. Moreover, in this case, the number of samples is actually the number of elements in the `lm_datasets` (in the version before your PR) as  `group_texts` changes the number of examples.

Therefore, all the preprocessing should be left as is here and the number of samples selected at the end.